### PR TITLE
Implement par_compute_global_potential

### PR DESCRIPTION
**Pull request by w1th0utnam3**
This implements a parallel version of `compute_global_potential` using Rayon with `map` + `try_reduce`.

### CODE REVIEW

File: tests/unit_tests/assembly/global.rs [hunk 1]
```diff
@@ -290,8 +290,10 @@ fn test_compute_global_potential() {
     }
 
     let global_potential = compute_global_potential(&MockScalarElementAssembler).unwrap();
-
     assert_scalar_eq!(global_potential, 9.0, comp = float);
+
+    let par_global_potential = par_compute_global_potential(&MockScalarElementAssembler).unwrap();
+    assert_scalar_eq!(par_global_potential, 9.0, comp = float);
 }
```
Comment (Andlon) [inline tests/unit_tests/assembly/global.rs:297]: Maybe move the mock assembler out of the test and put the parallel computation in its own test? Would be useful in order to be able to quickly spot tests that deal with parallelism (i.e. the test name would have "par" in it)

**Comment by Andlon** [comment]
Looks good to go otherwise though!

File: tests/unit_tests/assembly/global.rs (hunk 1 above)
Comment (w1th0utnam3) [reply tests/unit_tests/assembly/global.rs:300]: Makes sense

**Comment by Andlon** [comment]
Looks good to me!

**Pull request merged**